File: python/largest_odd_number_in_string.py
class Solution:
    def largestOddNumber(self, num: str) -> str:
        for i in reversed(range(len(num))):
            if int(num[i]) % 2 == 0:
                continue
            return num[0 : i + 1]

        return ""


def test_largestOddNumber_case_1():
    # arrange
    num: str = "52"
    expected: str = "5"

    # act
    solution = Solution()
    actual = solution.largestOddNumber(num)

    # assert
    assert expected == actual


def test_largestOddNumber_case_2():
    # arrange
    num: str = "4206"
    expected: str = ""

    # act
    solution = Solution()
    actual = solution.largestOddNumber(num)

    # assert
    assert expected == actual


def test_largestOddNumber_case_3():
    # arrange
    num: str = "35427"
    expected: str = "35427"

    # act
    solution = Solution()
    actual = solution.largestOddNumber(num)

    # assert
    assert expected == actual

File: rust/src/code/largest_odd_number_in_string.rs
struct Solution {}

impl Solution {
    pub fn largest_odd_number(num: String) -> String {
        let num: Vec<char> = num.chars().collect();
        for i in (0..num.len()).rev() {
            if num[i] as u8 % 2 == 0 {
                continue;
            }
            return num[0..i + 1].iter().collect::<String>();
        }
        return "".to_string();
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_largest_odd_number_case_1() {
        // arrange
        let num: String = "52".to_string();
        let expected: String = "5".to_string();

        // act
        let actual = Solution::largest_odd_number(num);

        // assert
        assert_eq!(expected, actual);
    }
}
